src/Integration.Vsix/CFamily/FileConfig.cs

```diff
@@ -30,19 +30,36 @@ partial class CFamilyHelper
     {
         internal class FileConfig
         {
-            public static FileConfig TryGet(ProjectItem dteProjectItem, string absoluteFilePath)
+            public static FileConfig TryGet(ILogger logger, ProjectItem dteProjectItem, string absoluteFilePath)
             {
                 var vcProject = dteProjectItem.ContainingProject.Object as VCProject;
                 var file = dteProjectItem.Object as VCFile;
                 if (vcProject == null || file == null)
                 {
                     return null;
                 }
-
                 var vcConfig = vcProject.ActiveConfiguration;
+                var projectKind = vcConfig.ConfigurationType;
+                // Unknown projects represent all the unspported project type like makefile projects
+                if (projectKind == ConfigurationTypes.typeUnknown)
+                {
+                    logger.WriteLine(@"Project's ""Configuration type"" is not supported.");
+                    return null;
+                }
+                // "ClCompile" for source files and "ClCompile" for header files
+                if (file.ItemType != "ClCompile")
+                {
+                    logger.WriteLine($"File's \"Item type\" is not supported. File: '{absoluteFilePath}'");
```

Comment:
Might be useful to output the item type in the log message.

Reply:
The "Item type" is the field in the file option. From my quick look, I didn't find a way to get the exact string in the UI(There are many possibilities) and most of them are mapped to something generic or unknown. Printing the actual `file.ItemType` will not be understandable